homeassistant/components/overkiz/binary_sensor.py

```diff
@@ -99,6 +100,16 @@ class OverkizBinarySensorDescription(
         device_class=BinarySensorDeviceClass.VIBRATION,
         value_fn=lambda state: state == OverkizCommandParam.DETECTED,
     ),
+    # DomesticHotWaterProduction/WaterHeatingSystem
+    OverkizBinarySensorDescription(
+        key=OverkizState.IO_OPERATING_MODE_CAPABILITIES,
+        name="Energy Demand Status",
+        device_class=BinarySensorDeviceClass.HEAT,
+        value_fn=lambda state: cast(dict, state).get(
+            OverkizCommandParam.ENERGY_DEMAND_STATUS
+        )
+        == 1,
```

Comment:
Multiline lambdas should be avoided. Can you create a local `_get_value_for_io_operation_mode_capabilities` method and set `value_fn=_get_value_for_io_operation_mode_capabilities,`